cmd/stackdriver-prometheus-sidecar/main.go

```diff
@@ -272,6 +276,18 @@ func main() {
 	if *projectId == "" {
 		*projectId = getGCEProjectID()
 	}
+
+	if cfg.sdOpencensus {
+		sd, err := oc_stackdriver.NewExporter(oc_stackdriver.Options{ProjectID: *projectId})
+		if err != nil {
+			level.Error(logger).Log("msg", "Creating Stackdriver exporter failed", "err", err)
+			os.Exit(1)
+		}
+		defer sd.Flush()
```

Comment:
Won't this run at the end of this block? I assume you want it to run when the process is shut down instead?

Reply:
I believe `defer` gets executed when the function ends (which is `main()` in this case, so when the process exits) rather than at the end of the surrounding block.

Reply:
TIL. Thank you!